lte/cloud/go/sms_ll/sms_ll.go

```diff
@@ -0,0 +1,101 @@
+package sms_ll
+
+import (
+	"github.com/warthog618/sms"
+	"github.com/warthog618/sms/encoding/tpdu"
+	"time"
+)
+
+// Generate fully encoded SMS PDUs for delivery to a UE (MS). Will handle
+// encoding and chunking of messages as appropriate. We first generate TPDUs,
+// then RP-DATA headers, and finally CP-DATA headers, resulting in a set of
+// byte arrays that can be directly delivered to a UE (MS).
+// Inputs:
+// 	message: A UTF-8 string representing the SMS.
+//	from_num: A E.164 encoded source number.
+//	timestamp: The sender timestamp for the SMS (generally, use current server time)
+//	ref_base: The starting reference number for this set of SMS messages. Should be a counter per IMSI.
+// Outputs:
+//	- Array of byte array representing the set of fully-encoded CP-DATA(RP-DATA(TPDU)) messages generated
+//	- Error	(if any)
+func GenerateSMSDelivers(message string, from_num string, timestamp time.Time, ref_base uint8) ([][]byte, error) {
```

Comment:
Do we need to export anything other than this function from the package? Everything else seems like implementation details that a caller otuside this package should not see

Reply:
My thinking is we need to expose at least Unmarshal for all types so callers can decode specific headers, and I'm not sure why we wouldn't want to expose the types and their fields given there's useful information in them (e.g., delivery status, the encapsulated TPDU, etc.)

Reply:
Capturing our offline discussion: we're going to expose only a very simplified public API for this module, so these are all going to be non-exported now.